packages/material/scss/common/_forms.scss

```diff
@@ -94,4 +86,34 @@
     .k-picker-wrap .k-dateinput-wrap {
         box-shadow: none;
     }
+
+    .k-state-invalid,
+    .ng-invalid.ng-touched,
+    .ng-invalid.ng-dirty {
+
+        &.k-autocomplete,
+        > .k-dateinput-wrap,
+        > .k-dropdown-wrap,
+        > .k-picker-wrap,
+        > .k-multiselect-wrap,
+        > .k-numeric-wrap {
```

Comment:
This rule will create a *lot* of selectors (90, to be precise, because of the nested classes, 3 * 6 * 5).
Looking at the screenshots, they show some improvement with the MultiSelect, but the amount of styles required for that is too much, and the styles have no effect on most components. Perhaps there is some detail that is missing from the screenshots?

Reply:
As .k-state-invalid should be applied to .k-widget, the selectors should be correct. I have fixed the tested page moving the class name to the correct element.